testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCloneTableCompatibility.java

```diff
@@ -0,0 +1,381 @@
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDeltaLakeCloneTableCompatibility
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Inject
+    @Named("s3.server_type")
+    private String s3ServerType;
+
+    private AmazonS3 s3;
+
+    @BeforeMethodWithContext
+    public void setup()
+    {
+        super.setUp();
+        s3 = new S3ClientFactory().createS3Client(s3ServerType);
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, DELTA_LAKE_EXCLUDE_113, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testTableChangesOnShallowCloneTable()
+    {
+        String baseTable = "test_dl_base_table_" + randomNameSuffix();
+        String clonedTable = "test_dl_clone_tableV1_" + randomNameSuffix();
+        String directoryName = "databricks-tablechanges-compatibility-test-";
+        String changeDataPrefix = "/_change_data";
+        try {
+            onDelta().executeQuery("CREATE TABLE default." + baseTable +
+                    " (a_int INT, b_string STRING) USING delta " +
+                    "LOCATION 's3://" + bucketName + "/" + directoryName + baseTable + "'");
+            onDelta().executeQuery("INSERT INTO default." + baseTable + " VALUES (1, 'a')");
+            onDelta().executeQuery("CREATE TABLE default." + clonedTable +
+                    " SHALLOW CLONE default." + baseTable +
+                    " TBLPROPERTIES (delta.enableChangeDataFeed = true)" +
+                    " LOCATION 's3://" + bucketName + "/" + directoryName + clonedTable + "'");
+            onDelta().executeQuery("INSERT INTO default." + clonedTable + " VALUES (2, 'b')");
+
+            Set<String> cdfFilesPostOnlyInsert = getFilesFromTableDirectory(directoryName + clonedTable + changeDataPrefix);
+            // Databricks version >= 12.2 keep an empty _change_data directory
+            assertThat(cdfFilesPostOnlyInsert).hasSize(0);
+
+            onDelta().executeQuery("UPDATE default." + clonedTable + " SET a_int = a_int + 1");
+            Set<String> cdfFilesPostOnlyInsertAndUpdate = getFilesFromTableDirectory(directoryName + clonedTable + changeDataPrefix);
+            assertThat(cdfFilesPostOnlyInsertAndUpdate).hasSize(2);
+
+            ImmutableList<Row> expectedRowsClonedTableOnTrino = ImmutableList.of(
+                    row(2, "b", "insert", 1L),
+                    row(1, "a", "update_preimage", 2L),
+                    row(2, "a", "update_postimage", 2L),
+                    row(2, "b", "update_preimage", 2L),
+                    row(3, "b", "update_postimage", 2L));
+            // table_changes function from trino isn't considering `base table inserts on shallow cloned table` as CDF as of v422
```

Comment:
> `table_changes` function from trino isn't considering `base table inserts on shallow cloned table` as CDF as of `v422`

Why is it so?
cc @homar

Reply:
@findinpath have raised an issue for the same https://github.com/trinodb/trino/issues/21183

Reply:
add the TODO in the code

Reply:
> as of v422

It's different from the current version. I would recommend removing.